src/main/java/hanshyn/onlinebookstore/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,51 @@
+package hanshyn.onlinebookstore.repository;
+
+import hanshyn.onlinebookstore.model.Book;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert book into DB:" + book);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery(
+                    "SELECT b FROM Book b", Book.class).getResultList();
```

Comment:
![image](https://github.com/hanshyn/online-book-store/assets/63095184/4b525b37-1e19-4ff1-8f3f-2f97ec0ca7a9)
You can simplify your query